RELEASE.md

```diff
@@ -9,7 +9,19 @@ This file is copied to nebari-dev/nebari-docs using a GitHub Action. -->
 
 ---
 
-## Upcoming Release
+
+## Release 2024.3.1 - March 14, 2024
```

Comment:
@Adam-D-Lewis it should be 2024.3.2 here

Reply:
ah, good catch